fvp-psa-sp.mk

```diff
@@ -0,0 +1,44 @@
+FVP_USE_BASE_PLAT		?= y
+FVP_VIRTFS_ENABLE		?= y
+FVP_VIRTFS_AUTOMOUNT		?= y
+MEASURED_BOOT			?= y
+MEASURED_BOOT_FTPM		?= n
+TS_SMM_GATEWAY			?= y
+
+TF_A_FLAGS ?= \
+	BL32=$(OPTEE_OS_PAGER_V2_BIN) \
+	BL33=$(EDK2_BIN) \
+	PLAT=fvp \
+	SPD=spmd \
+	SPMD_SPM_AT_SEL2=0 \
+	ARM_SPMC_MANIFEST_DTS=$(ROOT)/build/fvp/spmc_manifest.dts
+
+include fvp.mk
+include trusted-services.mk
+
+OPTEE_OS_COMMON_EXTRA_FLAGS += \
+	CFG_SECURE_PARTITION=y \
+	CFG_CORE_SEL1_SPMC=y \
+	CFG_CORE_HEAP_SIZE=131072 \
+	CFG_DT=y \
+	CFG_MAP_EXT_DT_SECURE=y
+
+# SP_COMMON_FLAGS is passed to each SP
+SP_COMMON_FLAGS ?=
+SP_PSA_ITS_EXTRA_FLAGS ?=
+SP_PSA_PS_EXTRA_FLAGS ?=
+SP_PSA_CRYPTO_EXTRA_FLAGS ?=
+SP_PSA_ATTESTATION_EXTRA_FLAGS ?=
+SP_SMM_GATEWAY_EXTRA_FLAGS ?=
+
+# The boot order of the SPs is determined by the order of calls here. This is
```

Comment:
How is the order enforced? Is it supposed to be via the `$(eval OPTEE_OS_COMMON_EXTRA_FLAGS+=...)` line below? Then I don't think it works. The build order of the `ffa-$1-sp` targets will be non-deterministic with parallel build (`make -jN`) and as a result the content of `OPTEE_OS_COMMON_EXTRA_FLAGS` may change.
One option may be to introduce a boot order argument (1, 2, 3...) and use that in the target names and dependencies.

Reply:
Yes, it's supposed to be the `$(eval OPTEE_OS_COMMON_EXTRA_FLAGS+=...)`, I agree it's very fragile.
Internally we've been using this solution for a long time now and never had any issues with it (even in a parallel build). But I understand your concern, I'll dig a bit deeper into this to see if there's some magic that keeps the correct order or just pure luck 😄

Reply:
Looks good